src/main/java/org/checkerframework/specimin/FieldDeclarationsVisitor.java

```diff
@@ -0,0 +1,50 @@
+package org.checkerframework.specimin;
+
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.body.FieldDeclaration;
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.SimpleName;
+import com.github.javaparser.ast.visitor.VoidVisitorAdapter;
+import java.util.HashMap;
+import java.util.Map;
+import org.checkerframework.checker.signature.qual.ClassGetSimpleName;
+
+/**
+ * This visitor is designed to assist the UnsolvedSymbolVisitor in creating synthetic files for
+ * unsolved NameExpr instances by listing all the names of declared fields in the current input
+ * file. It's important to note that this visitor is intended to be used in conjunction with the
+ * UnsolvedSymbolVisitor, so both visitors will traverse the same Java file.
+ * Thus, @ClassGetSimpleName names for involved classes should be sufficient.
+ */
+public class FieldDeclarationsVisitor extends VoidVisitorAdapter<Void> {
+  /**
+   * A mapping of field names to the @ClassGetSimpleName name of the classes in which they are
+   * declared. Since inner classes can be involved, a map is used instead of a simple list.
+   */
+  Map<String, @ClassGetSimpleName String> fieldAndItsClass;
```

Comment:
nit: I'd prefer `fieldNameToClassNameMap` or something else that makes it obvious just from the name that this object is a map. The current name is okay, but I think it is ambiguous whether it is intended to represent a single field or many fields just from the name.